compiler_gym/envs/compiler_env.py

```diff
@@ -1109,148 +295,47 @@ def multistep(
-            )
-            observation_spaces = observations
-        if rewards is not None:
-            warnings.warn(
-                "Argument `rewards` of CompilerEnv.multistep has been renamed "
-                "`reward_spaces`. Please update your code",
-                category=DeprecationWarning,
-            )
-            reward_spaces = rewards
-
-        # Coerce observation spaces into a list of ObservationSpaceSpec instances.
-        if observation_spaces:
-            observation_spaces_to_compute: List[ObservationSpaceSpec] = [
-                obs
-                if isinstance(obs, ObservationSpaceSpec)
-                else self.observation.spaces[obs]
-                for obs in observation_spaces
-            ]
-        elif self.observation_space_spec:
-            observation_spaces_to_compute: List[ObservationSpaceSpec] = [
-                self.observation_space_spec
-            ]
-        else:
-            observation_spaces_to_compute: List[ObservationSpaceSpec] = []
-
-        # Coerce reward spaces into a list of Reward instances.
-        if reward_spaces:
-            reward_spaces_to_compute: List[Reward] = [
-                rew if isinstance(rew, Reward) else self.reward.spaces[rew]
-                for rew in reward_spaces
-            ]
-        elif self.reward_space:
-            reward_spaces_to_compute: List[Reward] = [self.reward_space]
-        else:
-            reward_spaces_to_compute: List[Reward] = []
-
-        # Perform the underlying environment step.
-        observation_values, reward_values, done, info = self.raw_step(
-            actions, observation_spaces_to_compute, reward_spaces_to_compute
-        )
-
-        # Translate observations lists back to the appropriate types.
-        if observation_spaces is None and self.observation_space_spec:
-            observation_values = observation_values[0]
-        elif not observation_spaces_to_compute:
-            observation_values = None
-
-        # Translate reward lists back to the appropriate types.
-        if reward_spaces is None and self.reward_space:
-            reward_values = reward_values[0]
-            # Update the cumulative episode reward
-            self.episode_reward += reward_values
-        elif not reward_spaces_to_compute:
-            reward_values = None
-
-        return observation_values, reward_values, done, info
-
-    def render(
-        self,
-        mode="human",
-    ) -> Optional[str]:
-        """Render the environment.
-
-        CompilerEnv instances support two render modes: "human", which prints
-        the current environment state to the terminal and return nothing; and
-        "ansi", which returns a string representation of the current environment
-        state.
-
-        :param mode: The render mode to use.
-        :raises TypeError: If a default observation space is not set, or if the
-            requested render mode does not exist.
         """
```

Comment:
I think this should be on `CompilerEnv`. Is the reason it isn't because it is also on `gym.Env`?

Reply:
The reason was that it is a part of `gym.Env`, but I added it in `CompilerEnv` as well. It has a type hint and the additional documentation.